databuilder/databuilder/task/neo4j_staleness_removal_task.py

```diff
@@ -166,13 +169,17 @@ def _decorate_staleness(self,
         :return:
         """
         if self.ms_to_expire:
-            return statement.format(staleness_condition=textwrap.dedent(f"""\
-            (target.publisher_last_updated_epoch_ms < (timestamp() - ${MARKER_VAR_NAME})
-            OR NOT EXISTS(target.publisher_last_updated_epoch_ms))"""))
-
-        return statement.format(staleness_condition=textwrap.dedent(f"""\
-        (target.published_tag <> ${MARKER_VAR_NAME}
-        OR NOT EXISTS(target.published_tag))"""))
+            condition = f"""(target.publisher_last_updated_epoch_ms < (timestamp() - ${MARKER_VAR_NAME})"""
+            condition = (condition + ")"
+                         if self.retain_data_with_no_publisher_metadata
+                         else condition + "\nOR NOT EXISTS(target.publisher_last_updated_epoch_ms))")
+            return statement.format(staleness_condition=condition)
+
+        condition = f"""(target.published_tag <> ${MARKER_VAR_NAME}"""
```

Comment:
instead of `<>`, can we make this `<`? `<>` doesn't make much sense - if you run staleness remover immediately following the ingestion but across a date boundary, you'll delete everything you just ingested. 

Technically this is bw-incompatible but I have a hard time believing anyone will mind. In the worst case, it'll delete less data than was planned.